main.tf

```
@@ -79,11 +99,13 @@ locals {
   #     }
   #   }
   # }
-
   _common_configs = {
     for module, files in local._root_module_yaml_decoded : module => lookup(files, var.common_config_file, {})
   }
 
+  # If we're SingleInstance, then default_tf_workspace_enabled is true. Otherwise, use given value.
+  _default_tf_workspace_enabled = !local._multi_instance_structure ? true : var.default_tf_workspace_enabled
```

```suggestion
  _default_tf_workspace_enabled = local._multi_instance_structure ? var.default_tf_workspace_enabled : true
```
IMO, negations like `!` can make the logic harder to understand. I'd avoid them were possible.

---

Agreed! This is a GREAT callout -- THANK YOU!